### PR TITLE
React/week2

### PR DESCRIPTION
**Pull request by semih1239**

### CODE REVIEW

File: react/week2/todo-app/src/App.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import './App.css';
 import getTodos from './getTodos';
-import React from "react";
+import React, { useEffect } from 'react';
 
 const todos = getTodos()
```
Comment (islanderupnorth) [inline react/week2/todo-app/src/App.js:5]: It's possible to "export" the "const todos" from the todos file without wrapping it in a function like so:

export const todos = []

**Review by islanderupnorth — COMMENTED** [review]
Overall it's good to see you properly understand how re-rendering triggers updates in React!

A good practice is to use the useState as the "brain" of the logic and not to add Components to it but to store the "long list of the todos". Then in the final HTML return in your App() have the components which are affected by the the new long list in the useState to change.

Just like you used the "todos.filter(todo => todo.key != id)" logic you can do that in the JSX/HTML with the list.

Brief example of what I mean:

```
function App() {
const [todos, setToDos] = useState([LIST_OF_TODOS])

function deleteTodoItem(id) {
    setTodos( todos.filter(todo => todo.key !== id) )
}

return (
 <ul>
   {todos.map((todo) => {
      return <ToDo description={todo.description} deleteTodo={() => deleteToDoItem(todo.id)} key={todo.randomTextKey} />}
</ul>
)
}

```

File: react/week2/todo-app/src/App.js (hunk 1 above)
Comment (semih1239) [reply react/week2/todo-app/src/App.js:5]: I didnt know that. Thank you 👍

**Comment by semih1239** [comment]
> Overall it's good to see you properly understand how re-rendering triggers updates in React!
> 
> A good practice is to use the useState as the "brain" of the logic and not to add Components to it but to store the "long list of the todos". Then in the final HTML return in your App() have the components which are affected by the the new long list in the useState to change.
> 
> Just like you used the "todos.filter(todo => todo.key != id)" logic you can do that in the JSX/HTML with the list.
> 
> Brief example of what I mean:
> 
> ```
> function App() {
> const [todos, setToDos] = useState([LIST_OF_TODOS])
> 
> function deleteTodoItem(id) {
>     setTodos( todos.filter(todo => todo.key !== id) )
> }
> 
> return (
>  <ul>
>    {todos.map((todo) => {
>       return <ToDo description={todo.description} deleteTodo={() => deleteToDoItem(todo.id)} key={todo.randomTextKey} />}
> </ul>
> )
> }
> ```

Thank you very much for your feedback. I will update my homework with your feedbacks 🙏

**Pull request merged**